### PR TITLE
feat: Add support for product imports

### PR DESCRIPTION
**Pull request by sradevski**
Depends on #8281 (will retarget to develop once that is merged)

Designs: https://www.figma.com/design/z3aUuOVWUKmdHH0ofmMpEV/Web-app-3.0?node-id=6612-67477&t=FQjXXVUyAbMvFhfh-0

CLOSES CC-253

### CODE REVIEW

File: packages/admin-next/dashboard/src/components/common/file-preview/file-preview.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+              <ArrowDownTray />
+            </a>
+          </IconButton>
+        )}
+      </div>
+    </div>
+  )
+}
+
+const FileThumbnail = () => {
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/components/common/file-preview/file-preview.tsx:60]: This is not really an icon, so I wasn't sure how to handle it. I think having it defined here is fine, but let me know if another place would be better.

File: packages/admin-next/dashboard/src/routes/products/product-import/product-import.tsx [hunk 2]
```diff
@@ -15,7 +21,119 @@ export const ProductImport = () => {
...
+          {t("products.import.template.title")}
+        </Heading>
+        <Text size="small" className="text-ui-fg-subtle">
+          {t("products.import.template.description")}
+        </Text>
+        <div className="mt-4">
+          <FilePreview
+            filename={"medusa-template-product-list.csv"}
+            // TODO: Where would the template file be stored?
+            url={"https://example.com"}
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/routes/products/product-import/product-import.tsx]: @olivermrbl where did we store the template for the imports before?

Comment (olivermrbl) [reply packages/admin-next/dashboard/src/routes/products/product-import/product-import.tsx]: We constructed and downloaded the file on the fly:
https://github.com/medusajs/medusa/blob/v1.x/packages/admin-ui/ui/src/domain/products/batch-job/download-template.ts

Comment (sradevski) [reply packages/admin-next/dashboard/src/routes/products/product-import/product-import.tsx]: Ok cool, let's do that, seems like the simpler option anyway.

Comment (sradevski) [reply packages/admin-next/dashboard/src/routes/products/product-import/product-import.tsx]: Done

**Review by riqwan — APPROVED** [review]
lgtm 🤞🏻

**Pull request merged**